### PR TITLE
Make framework exceptions nonfatal.

### PR DESCRIPTION
**Pull request by dham**
If the framework munging fails for any reason, log the exception
but keep calm and carry on.

### CODE REVIEW

File: scripts/firedrake-install [hunk 1]
```diff
@@ -1184,7 +1184,12 @@ os.environ["VIRTUAL_ENV"] = virtual_env_vars["base"]
 
 os.chdir(firedrake_env)
 
-framework = fix_osx_virtualenv_python()
+try:
+    framework = fix_osx_virtualenv_python()
+except BaseException as e:
+    log.exception(e)
+    framework = False
```
Comment (wence-) [inline scripts/firedrake-install]: This is OK, although I wonder, depending on which step fails, whether we can end up in a broken state but still say "success"?

Comment (dham) [inline scripts/firedrake-install]: Hmm. That's also possible.

**Comment by miklos1** [comment]
Rebased on `master`.

**Comment by wence-** [comment]
Alternate proposal, we could remove this code entirely, people can still use matplotlib, they just need to select a different backend (as described in the matplotlib faq).  I do that in 3ef94216951.

**Comment by wence-** [comment]
replaced by removing framework munging.

**Pull request closed without merging**